mmv1/third_party/terraform/services/gkehub2/resource_gke_hub_feature_test.go.erb

```diff
@@ -347,6 +347,143 @@ resource "google_gke_hub_feature" "feature" {
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		CheckDestroy:             testAccCheckGKEHubFeatureDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccGKEHubFeature_FleetDefaultMemberConfigServiceMesh(context),
+			},
+			{
+				ResourceName:            "google_gke_hub_feature.feature",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"project"},
+			},
+			{
+				Config: testAccGKEHubFeature_FleetDefaultMemberConfigServiceMeshUpdate(context),
+			},
+			{
+				ResourceName:      "google_gke_hub_feature.feature",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccGKEHubFeature_FleetDefaultMemberConfigServiceMesh(context map[string]interface{}) string {
+	return gkeHubFeatureProjectSetupForGA(context) + acctest.Nprintf(`
+resource "google_project_service" "test_mesh" {
+  service = "meshconfig.googleapis.com"
+}
+
+resource "google_project_service" "test_mci" {
+  service = "multiclusteringress.googleapis.com"
+}
+
+resource "google_project_service" "test_acm" {
+  service = "anthosconfigmanagement.googleapis.com"
+}
+
+resource "google_project_service" "test_mcsd" {
+  service = "multiclusterservicediscovery.googleapis.com"
+}
+
+resource "google_project_service" "test_compute" {
+  service = "compute.googleapis.com"
+  disable_on_destroy = false
+}
+
+resource "google_project_service" "test_container" {
+  service = "container.googleapis.com"
+  disable_on_destroy = false
+}
+
+resource "google_project_service" "test_gkehub" {
+  service = "gkehub.googleapis.com"
+  disable_on_destroy = false
+}
+
+resource "google_gke_hub_membership" "membership" {
+  membership_id = "tf-test%{random_suffix}"
+}
+
+resource "google_gke_hub_feature" "feature" {
+  name = "servicemesh"
+  location = "global"
+  fleet_default_member_config {
+    mesh {
+      management = "MANAGEMENT_AUTOMATIC"
+    }
+  }
+  depends_on = [google_project_service.test_mesh]
+}
+`, context)
+}
+
+func testAccGKEHubFeature_FleetDefaultMemberConfigServiceMeshUpdate(context map[string]interface{}) string {
+	return gkeHubFeatureProjectSetupForGA(context) + acctest.Nprintf(`
+resource "google_project_service" "test_mesh" {
+  service = "meshconfig.googleapis.com"
+}
+
+resource "google_project_service" "test_mci" {
+  service = "multiclusteringress.googleapis.com"
+}
+
+resource "google_project_service" "test_acm" {
+  service = "anthosconfigmanagement.googleapis.com"
+}
+
+resource "google_project_service" "test_mcsd" {
+  service = "multiclusterservicediscovery.googleapis.com"
+}
+
+resource "google_project_service" "test_compute" {
+  service = "compute.googleapis.com"
+  disable_on_destroy = false
+}
+
+resource "google_project_service" "test_container" {
+  service = "container.googleapis.com"
+  disable_on_destroy = false
+}
+
+resource "google_project_service" "test_gkehub" {
```

Comment:
For this test to work in VCR REPLAYING mode you will need to chain all of these google_project_service resources together via depends_on.